packages/monaco/src/browser/monaco-workspace.ts

```diff
@@ -144,32 +145,46 @@ export class MonacoWorkspace extends BaseMonacoWorkspace implements lang.Workspa
         };
     }
 
-    applyEdit(changes: lang.WorkspaceEdit): Promise<boolean> {
+    async applyEdit(changes: lang.WorkspaceEdit): Promise<boolean> {
         const workspaceEdit = this.p2m.asWorkspaceEdit(changes);
-        const promises = [];
-        for (const edit of workspaceEdit.edits) {
-            promises.push(this.textModelService.createModelReference(edit.resource).then(reference => {
-                const model = reference.object.textEditorModel;
-                // start a fresh operation
-                model.pushStackElement();
-                const range = edit.range;
-                const selections = [new monaco.Selection(range.startLineNumber, range.startColumn, range.endLineNumber, range.endColumn)];
-                model.pushEditOperations(selections, [{
+        const uri2Edits = this.groupEdits(workspaceEdit);
+        for (const uri of uri2Edits.keys()) {
+            const editorWidget = await this.editorManager.open(new URI(uri));
+            const editor = get(editorWidget);
+            if (editor) {
+                const model = editor.document.textEditorModel;
+                const currentSelection = new monaco.Selection(
```

Comment:
`const currentSelections = editor.editor.getSelections();`

Reply:
`editor.editor` is protected

Reply:
sorry, I meant `editor.getControl().getSelections()`